src/pip/_internal/utils/subprocess.py

```diff
@@ -132,6 +132,8 @@ def call_subprocess(
             cwd=cwd,
             env=env,
             errors="backslashreplace",
+            # to avoid credential prompts through terminal device
+            start_new_session=True,
```

Comment:
Are you this is safe if `--no-input` is not passed? In the normal case, we still want the prompts from the underlying Git/SSH calls to passthrough. I'm not familiar with the `start_new_session=True` argument.

Reply:
Oh.. you're right. In that case, I'm not sure what to do. You were right, it seems like this can only be properly fixed by a timeout on tests.
Is this PR still eligible for merging, at least only the bugfix without the test ?

Reply:
I'd like to merge it, but I wasn't sure about this specific change as it could have a global impact :)

I'll into adding timeout support to our testing framework at some point. The test can stay, we'll just have to be careful to not regress (which should be easy as we almost never touch the VCS code these days). :+1:

Reply:
Ok ! I just removed my last commit with `start_new_session`